### PR TITLE
Add average win percentage and tests using gameteams data

### PR DESCRIPTION
**Pull request by dunlapww**
Add average win percentage methods and tests using gameteam data

All tests passing

### CODE REVIEW

**Review by jdcarey128 — CHANGES_REQUESTED** [review]
Looks like a couple quick merge conflict resolutions, but then you're good.

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -381,4 +395,12 @@ def fewest_tackles(season)
   end
 
 # ~~~ TEAM METHODS~~~
+
+  def avg_win_perc(id)
+    find_percent(total_wins(filter_by_teamid(id)), total_game_teams(filter_by_teamid(id)))
+  end
```
Comment (jdcarey128) [inline lib/stat_tracker.rb]: I think we should name methods the same as they are in the spec - just in case that causes issues with the spec harness.

**Comment by jakeheft** [comment]
Will, for your merge conflicts, it looks like you did not pull from the latest main (from my PR that was merged last night). If you do that, you should be good to go on here.

**Review by jdcarey128 — APPROVED** [review]
Thanks for the updates. Looks good.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -212,6 +212,18 @@ def team_id_to_team_name(id)
...
+    game_teams_filtered.count do |gameteam|
+      gameteam.result == "WIN"
+    end
+  end
+
+  def filter_by_teamid(id)
+    @game_teams.select do |game_team|
+      game_team.team_id == id
+    end
+  end
```
Comment (jakeheft) [inline lib/stat_tracker.rb:225]: Be specific with id argument; change to team_id (as opposed to home_id or away_id)

**Review by jakeheft — APPROVED** [review]

**Pull request merged**